lib/eventemitter2.js

```diff
@@ -882,42 +1282,89 @@
       var ns = typeof type === 'string' ? type.split(this.delimiter) : type.slice();
       var leafs = searchListenerTree.call(this, null, ns, this.listenerTree, 0);
 
-      for (var iLeaf=0; iLeaf<leafs.length; iLeaf++) {
+      for (var iLeaf = 0; iLeaf < leafs.length; iLeaf++) {
         var leaf = leafs[iLeaf];
         leaf._listeners = null;
       }
-    }
-    else if (this._events) {
+    } else if (this._events) {
       this._events[type] = null;
     }
     return this;
   };
 
-  EventEmitter.prototype.listeners = function(type) {
-    if (this.wildcard) {
-      var handlers = [];
-      var ns = typeof type === 'string' ? type.split(this.delimiter) : type.slice();
-      searchListenerTree.call(this, handlers, ns, this.listenerTree, 0);
-      return handlers;
-    }
+  EventEmitter.prototype.listeners = function (type) {
+    var _events = this._events;
+    var keys, listeners, allListeners;
+    var i;
+    var listenerTree;
 
-    this._events || init.call(this);
+    if (type === undefined) {
+      if (this.wildcard) {
+        throw Error('event name required for wildcard emitter');
```

Comment:
Is this a breaking change?

Reply:
No. In the previous version, this method required an event type, in this PR it became optional, and now we can collect all the listeners, not just a specific type. Wildcard events without specifying an event type are not currently supported. So this is not a breaking change.